lib/wibox/layout/margin.lua

```diff
@@ -78,6 +78,14 @@ function margin:get_children()
     return {self.widget}
 end
 
+--- Replace the layout children
+-- This layout only accept one children, all others will be ignored
+-- @tparam table children A table composed of valid widgets
+function margin:set_children(children)
+    self.widget = children and children[1]
```

Comment:
Should all of these functions use `:check_widget()`? Actually... these could just use `:set_widget(children and children[1])` and then there is only one function doing all of this. Similarly, `fixed` could use `self:reset() self:add(unpack(children or {}))`.

Am I micro-optimizing for no real gain? (Does `check_widget()` make sense or am I just scared from when I started this widget stuff and had tons of errors while drawing, because invalid widgets were set?)